### PR TITLE
loader.c: fill_rsp - Fix possible array index overflow - L1 CERT-C

### PR DESCRIPTION
**Pull request by DOAR-Infineon**

### CODE REVIEW

File: boot/bootutil/src/loader.c [hunk 1]
```diff
@@ -182,6 +182,11 @@ fill_rsp(struct boot_loader_state *state, struct boot_rsp *rsp)
             break;
         }
     }
+    /* At least one image must be active, otherwise skip the execution */
+    if(BOOT_CURR_IMG(state) >= BOOT_IMAGE_NUMBER)
```
Comment (d3zd3z) [inline boot/bootutil/src/loader.c:186]: Should be a space between the 'if' and the '('.

File: boot/bootutil/src/loader.c [hunk 2]
```diff
@@ -182,6 +182,11 @@ fill_rsp(struct boot_loader_state *state, struct boot_rsp *rsp)
...
+    {
```
Comment (d3zd3z) [inline boot/bootutil/src/loader.c:187]: Brace should be at the end of the previous line, preceded by a space.

**Review by d3zd3z — CHANGES_REQUESTED** [review]

**Review by de-nordic — APPROVED** [review]

**Comment by d3zd3z** [comment]
The commits will need Signed off by lines. I recommend the author using their real name in the Author field. Let us know if you need help with this, as it is a fairly simple change that one of the maintainers could do, but we do need your permission to fix the author email address and add the signed-off-by.

**Review by d3zd3z — CHANGES_REQUESTED** [review]
These should probably just be squashed as well.

File: boot/bootutil/src/loader.c (hunk 1 above)
Comment (DOAR-Infineon) [reply boot/bootutil/src/loader.c]: Fixed

File: boot/bootutil/src/loader.c (hunk 2 above)
Comment (DOAR-Infineon) [reply boot/bootutil/src/loader.c]: Fixed

**Review by d3zd3z — APPROVED** [review]

**Pull request merged**